internal/it/util.go

```diff
@@ -542,3 +588,68 @@ func sortedString(b []byte) string {
 	s = strings.ReplaceAll(s, "\r", "")
 	return s
 }
+
+var nextPort int32 = 10000
+
+func NextPort() int {
+	// let minimum step be 10, just a round, safe value.
+	// note that some tests may not use the MemberCount() function for the cluster size.
+	const maxStep = 10
+	step := MemberCount()
+	if step < maxStep {
+		step = maxStep
+	}
+nextblock:
+	for {
+		start := int(atomic.AddInt32(&nextPort, int32(step))) - step
+		// check that all ports in the range are open
+		for port := start; port < start+step; port++ {
+			if !isPortOpen(port) {
```

Comment:
If desired port is already engaged, we can look for the next available one until there it is allocated.